apps/aesophia/test/contracts/erc20_token.aes

```diff
@@ -0,0 +1,97 @@
+contract ERC20Token =
+  record state = {
+    totalSupply : int,
+    decimals : int,
+    name : string,
+    symbol : string,
+    balances : map(address, int),
+    allowed : map(address, map(address,int)),
+    // Logs, remove when native Events are there
+    transfer_log : list((address,address,int)),
+    approval_log : list((address,address,int))}
+
+  // init(100000000, 10, "Token Name", "TKN")
+  public stateful function init(_totalSupply : int, _decimals : int, _name : string, _symbol : string ) = {
```

Comment:
I would expect [no stateful keyword for init](https://github.com/aeternity/protocol/blame/99e569126aa9a2b2be9953e6a0461c6b0258cdbd/contracts/sophia.md#L102-L104).

Reply:
And I expect the state record definition can be used as a type. So I understand the most explicit definition would be:
```
public function init(...) : state = {
```

Reply:
And it looks like [`: state` may be implied for init](https://github.com/aeternity/protocol/blame/99e569126aa9a2b2be9953e6a0461c6b0258cdbd/contracts/sophia.md#L101).

Reply:
Tracked https://www.pivotaltracker.com/story/show/160750737